### PR TITLE
Allow edit groups from an agent

### PR DESCRIPTION
**Pull request by lucianogorza**
### Description
In order to allow users to edit the groups from an agent, we have to introduce the following changes:
- Modify the agent's table actions to add the option `Edit groups`.
- Add a new generic component `WzElementPermissions` to manage permission when we need a component different to a button.
- Migrate the `AgentsTable` class component to a functional component because we need to use an existing hook to disable the `Edit groups` action button.
 
### Issues Resolved
https://github.com/wazuh/wazuh-dashboard-plugins/issues/6138

#### Actions menú
image

#### Actions menú when never connected
image
image

#### `Edit groups` modal
image

#### Actions menú with read only user
image
image

#### `Deploy new agent` button with readonly user
image

### Test
- Go to Endpoints Summary
- Verify that actions work consistently with agent state and user permissions.
- Test buttons that have permission validations
- Test table filters
To create a readonly user: https://documentation.wazuh.com/current/user-manual/user-administration/rbac.html#creating-and-setting-a-wazuh-read-only-user

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: plugins/main/public/components/endpoints-summary/table/agents-table.tsx [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+    {
+      field: 'group_config_status',
+      name: 'Synced',
+      sortable: true,
+      show: false,
+      render: synced => <AgentSynced synced={synced} />,
+      searchable: true,
+    },
+    {
+      name: 'Actions',
```
Comment (jbiset) [inline plugins/main/public/components/endpoints-summary/table/agents-table.tsx:227]: In order to hide/show the actions column, the `field: 'actions'` must be specified.

**Review by jbiset — COMMENTED** [review]
## CR :green_circle:  (See comments)

## Test

### Actions menú :green_circle:
Evidence_1

### Actions menú when never connected :green_circle:
Evidence_2

Evidence_2_B

### Edit groups modal :green_circle:
Evidence_3


### Actions menú with read only user :green_circle:
Evidence_4


### Deploy new agent button with read only user :green_circle:
Evidence_5

### Verify that actions work consistently with agent state and user permissions :green_circle: 

### Test buttons that have permission validations :green_circle: 

### Test table filters :green_circle: 

### Test table add/remove columns :yellow_circle: 
Cannot change visibility of `actions` column

[Issue_actions.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/8fff9432-a8d5-4d6d-8721-aa2d68338407)

File: plugins/main/public/components/endpoints-summary/table/agents-table.tsx (hunk 1 above)
Comment (lucianogorza) [reply plugins/main/public/components/endpoints-summary/table/agents-table.tsx:228]: Done

**Review by jbiset — APPROVED** [review]

**Review by asteriscos — APPROVED** [review]
## Test :green_circle:  (See :yellow_circle:  comments)

## Test

### Actions menú :green_circle:
image


### Actions menú when never connected :yellow_circle: 
image

> [!Note]
> I'm not sure if it makes any sense to allow to see the Agent Details and then show the "Agent never connected" message. Even more when there's no quick way to go back.

image

image


### Edit groups modal :green_circle:

image


### Actions menú with read-only user :green_circle:


### Deploy new agent button with read-only user :green_circle:

image


### Verify that actions work consistently with agent state and user permissions :green_circle: 

image


### Test buttons that have permission validations :green_circle: 

### Test table filters :green_circle:

**Pull request merged**